Misc fixes for baremetal install script

bpradipt opened the pull request.

bpradipt committed:
Create constants for TDX and SNP node label

This will make it easier to reuse these labels when installing other
K8s manifests.

Drive-by-fixes: formatting fixes

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

bpradipt committed:
Fix shellcheck warnings

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

bpradipt committed:
Bail out early if TDX and SNP labels are missing

If installing with TEE type, then bail out early if
the required node labels are missing

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

bpradipt committed:
Use TDX label for all TDX setup

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

bpradipt committed:
Fix nodeSelector spec

It needs to be added as part of the container spec

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

bpradipt committed:
Install TDX prereq after node labelling

This avoids installing prerequisites when not needed

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

bpradipt committed:
Add tolerations to pccs to enable deployment on master

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

bpradipt committed:
Change install order

This is necessitated to target regular clusters
with worker nodes.

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

fidencio commented on scripts/install-helpers/baremetal-coco/intel-dcap/registration-ds.yaml.in:
```diff
@@ -24,7 +24,7 @@ spec:
     spec:
       nodeSelector:
-        intel.feature.node.kubernetes.io/sgx: 'true'
+        intel.feature.node.kubernetes.io/tdx: 'true'
```
It actually has to be `sgx`  here, as we do depend on SGX to do the node registration.

fidencio commented on scripts/install-helpers/baremetal-coco/intel-dcap/qgs.yaml:
```diff
@@ -43,6 +41,8 @@ spec:
         sgx.intel.com/quote-provider: tdx-qgs
     spec:
+      nodeSelector:
+        intel.feature.node.kubernetes.io/tdx: 'true'
```
This one must be added, yes.

fidencio reviewed (approved):
lgtm, thanks @bpradipt!

One nit that I'd add here, and you can decide whether you want to address it here or not, but 900 seconds is not enough for muilti-node, making the user having to rerun the script.

I'd bump it to a really high number, like 2700 (45 minutes) to ensure that the deployment can actually finish within the time.

cclaudio reviewed (approved):
LGTM.

bpradipt committed:
Increase default timeout value

Usually multi-node cluster takes quite some
time to finish. Hence increase the default timeout.

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

bpradipt committed:
Fix detection of SNO or converged cluster

An SNO or converged cluster doesn't have any nodes
in the `worker` MachineConfigPool. All nodes are part
of the `master` MachineConfigPool. We use this aspect
to detect SNO or converged cluster to install Kata with CoCo
support.

A true multi-node cluster with have at least one node in the
`worker` MachineConfigPool.
For true multi-node cluster, we expect the admin to label the
target worker node for CoCo installation with specific label (eg.
coco-bm=true) and specify this label in the BM_NODE_LABEL env variable
(eg. export BM_NODE_LABEL="coco-bm=true") before starting the
installation

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>
